project_setup.py

```diff
@@ -0,0 +1,181 @@
+    return "/".join(Path(repo_url).parts[-2:]).replace(".git", "")
+
+
+def run_setup(
+    repo_url:str,
+    package_name:str,
+    repo:git.Repo = git.Repo(),
+) -> None:
+    """Helper to setup a new project based on this template.
+
+    It does the following things:
+    1. Updates the pyproject.toml and README.md files
+    2. Renames the package_name directory to a python safe version of the
+       repo name
+    3. Updates the package_name and <package_name> in all files to a python safe
+       version of the repo name
+
+    Args:
+        repo_url (str): The URL of the repo
+        package_name (str): The name of the package
+        repo (git.Repo): The git repo to use
+
+    Returns:
+        None
+    """
+    # Step 1: Update the pyproject.toml and README.md ==========================
+    rtd_guess = rtd_project_guess(repo_url)
+    codecov_guess = codecov_project_guess(repo_url)
+
+    pyproject_toml_replacement_pairs = [
+        (TEMPLATE_REPO_URL, repo_url),
+    ]
+    if replace_file_contents(
+        pyproject_toml_replacement_pairs,
+        Path(repo.working_dir) / Path("pyproject.toml"),
+    ):
+        print(UPDATED_PYPROJECT_TOML)
+
+    readme_replacement_pairs = [
+        (TEMPLATE_REPO_URL, repo_url),
+        (TEMPLATE_RTD, rtd_guess),
+        (TEMPLATE_CODECOV, codecov_guess),
+    ]
+    if replace_file_contents(
+        readme_replacement_pairs,
+        Path(repo.working_dir) / Path("README.md"),
+    ):
+        print(UPDATED_README)
+
+    # Step 2: Rename the package_name directory ================================
+    repo.git.mv(TEMPLATE_PACKAGE_NAME, package_name)
+    repo.git.commit("-m", f"renamed dir {TEMPLATE_PACKAGE_NAME} to {package_name}")
+    print(UPDATED_DIR.format(old_dir=TEMPLATE_PACKAGE_NAME, new_dir=package_name))
+
+    # Step 3: Update the package_name and <package_name> in all files ==========
+
+    # filter out files that we don't want to modify
+    files_to_check = list(filter(
+        valid_file_predicate,
+        map(lambda f: Path(repo.working_dir) / Path(f), repo.git.ls_tree("-r", "--name-only", "HEAD").split("\n")),
+    ))
+
+    # order matters, replace the bracketed version first
+    package_name_replacement_pairs = [
+        ("<" + TEMPLATE_PACKAGE_NAME + ">", package_name),
+        (TEMPLATE_PACKAGE_NAME, package_name),
+    ]
+    for file_path in files_to_check:
+        # only print positive results
+        if replace_file_contents(package_name_replacement_pairs, file_path):
+            print(UPDATED_FILE.format(package_name=package_name, file_path=file_path))
+
+    self_destruct(Path(repo.working_dir))
+
+    repo.git.commit("-am", f"updated package name to {package_name} in files")
+
+    print(SETUP_COMPLETE.format(package_name=package_name))
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--repo_url", type=str, default="")
+    parser.add_argument("--package_name", type=str, default="")
```

Comment:
``help`` messages would be good here to disambiguate between these representing the existing repo or the desired new repo (the latter being correct here).

See [help](https://docs.python.org/3/library/argparse.html#help)

Reply:
When this is run it will be in the new repo. So, the repo url we get guess will generally be right. The package name, though I imagine it would be closely related to the url, may be different.  I'll add the help messages though.